src/ui/profile/AddProfile.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/profile.module.css';
+import ProfileModel from '../../models/profile';
+import type from '../../utils/types';
+
+const BEYOND_ID_LENGTH = 8;
+
+function AddProfileForms(props) {
+  const { onAddProfile, onCancel, onError } = props;
+  const [charId, setCharId] = useState('');
+
+  const validateInputId = (input) => {
+    const parsedId = parseInt(input, 10);
+    if (input.length !== BEYOND_ID_LENGTH || !type.isNumber(parsedId)) {
+      throw new Error('Invalid DnD Beyond Character ID.');
+    }
+    return parsedId;
+  };
+
+  const handleAddCharacter = () => {
+    let id;
+    try {
+      id = validateInputId(charId);
+    } catch (err) {
+      onError(err.message);
+      return;
+    }
+
+    const url = `https://www.dndbeyond.com/character/${id}/json`;
+
+    try {
+      onAddProfile(id, fetch(url)
+        .then((response) => response.json())
+        .catch(() => Promise.reject(new Error('Profile not found.')))
+        .then((jsonData) => {
+          if (jsonData.errorCode === 404) {
+            const strs = [
+              'Could not retrieve profile.',
+              'Please verify your character privacy is set to public.',
+            ];
+            throw new Error(strs.join('\n'));
+          }
+          const avatar = jsonData.avatarUrl || '';
+          const level = jsonData.classes.reduce((total, classObj) => {
+            const classLevel = parseInt(classObj.level, 10);
+            return total + classLevel;
+          }, 0);
+          return ProfileModel({
+            id,
+            name: jsonData.name,
+            avatar,
+            level,
+          });
+        }));
+    } catch (err) {
+      if (err.message === 'ALREADY_EXISTS') {
+        onError('Character already added.');
+      } else {
+        throw err;
+      }
+    }
+  };
+
+  return (
+    <div>
+      <form className={styles.InputForm}>
+        <label htmlFor="idInput">
+          Character ID:
+          <input
+            id="idInput"
+            type="text"
+            minLength="8"
```

Comment:
There's really no reason to set a `minLength` on an input because you're doing validation yourself. This is another one of those things that existed in old HTML forms and isn't really used anymore.

`maxLength` carries the extra effect of actually restricting the user from typing any more characters, so it is still useful.

Reply:
I think if the input is less than `minLength`, the element gets a red border. I would code that myself but i'm lazy 😄